commerce-payment-service/src/main/java/com/liferay/commerce/payment/service/impl/CommercePaymentMethodGroupRelLocalServiceImpl.java

```diff
@@ -279,6 +308,77 @@ public CommercePaymentMethodGroupRel setActive(
 
 		commercePaymentMethodGroupRel.setActive(active);
 
+		if (!active) {
+			Map<String, CommercePaymentMethod> commercePaymentMethodsMap =
+				_commercePaymentMethodRegistry.getCommercePaymentMethods();
+
+			List<CommercePaymentMethod> recurringCommercePaymentMethods =
+				new ArrayList<>();
+
+			for (CommercePaymentMethod commercePaymentMethod :
+					commercePaymentMethodsMap.values()) {
+
+				if (commercePaymentMethod.isProcessRecurringEnabled()) {
+					recurringCommercePaymentMethods.add(commercePaymentMethod);
+				}
+			}
+
+			if (recurringCommercePaymentMethods.size() == 1) {
+				CommercePaymentMethod lastRecurringCommercePaymentMethod =
+					recurringCommercePaymentMethods.get(0);
+
+				String lastRecurringCommercePaymentMethodKey =
+					lastRecurringCommercePaymentMethod.getKey();
+
+				if (!lastRecurringCommercePaymentMethodKey.equals(
+						commercePaymentMethodGroupRel.getEngineKey())) {
+
+					return commercePaymentMethodGroupRelPersistence.update(
+						commercePaymentMethodGroupRel);
+				}
+
+				SearchContext searchContext = new SearchContext();
+
+				Map<String, Serializable> attributes = new HashMap<>();
+
+				attributes.put(Field.STATUS, WorkflowConstants.STATUS_APPROVED);
+
+				attributes.put(CPField.PUBLISHED, true);
+				attributes.put(CPField.SUBSCRIPTION_ENABLED, true);
+
+				long groupId = commercePaymentMethodGroupRel.getGroupId();
+
+				attributes.put("commerceChannelGroupId", groupId);
+
+				searchContext.setAttributes(attributes);
+
+				searchContext.setCompanyId(
+					commercePaymentMethodGroupRel.getCompanyId());
+
+				CPDataSourceResult cpDataSourceResult =
```

Comment:
@alecsloan why we need this?

Reply:
If the admin is disabling the last recurring payment method we disable subscriptions on all products where subscription is enabled.